### PR TITLE
#115 endpoint routes

### LINKED ISSUE #115: Endpoint logic clutters main server code
Endpoints should be moved to their own dedicated file. This will make it easier to parse the endpoints at a glance.

### PR DESCRIPTION
**Pull request by wlthomson**
Fixes #115.

### CODE REVIEW

**Comment by wlthomson** [comment]
Tried to merge with development but getting eslint errors for `RequisitionsPage.js` and `CreateRequisitionModal.js`. Once these are resolved, will merge.

**Comment by wlthomson** [comment]
Eslint complaining about `propTypes` resolved.

File: src/server/routes/masterlist.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/* eslint-disable no-console */
+
+export const getMasterlist = ({ db, secret }) => async (req, res) => {
+  const token = req.getJWT();
+  if (token && token.isValid(secret)) {
+    const storeId = token.getField('storeId');
+
+    try {
+      const masterlists = await db.getMasterLists(storeId);
```
Comment (joshxg) [inline src/server/routes/masterlist.js]: Mentioned in #137 but will do so here also - I think this should be getMasterlists

Comment (wlthomson) [reply src/server/routes/masterlist.js]: Thanks, will fix in #137.

File: src/server/database/index.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import { createTestRecords } from '../test/test';
+
+export const initialiseDatabase = async path => {
+  const dbFile = `${path}/${dbName}`;
+
+  const dbExists = fs.existsSync(dbFile);
+
+  const db = await new Database(dbFile, dbSchema).initialise();
+
+  // If db file exists, do not add test records.
```
Comment (joshxg) [inline src/server/database/index.js]: Sorry to be annoying but can we remove these newlines? Also this comment is pretty redundant?

File: src/server/middleware/middleware.js [hunk 3]
```diff
@@ -1,4 +1,7 @@
-import JWT from '../authorisation/JWT';
+import bodyParser from 'body-parser';
+import cors from 'cors';
+
+import JWT from '../../utilities/authorisation/JWT';
 
 const logRequest = req => {
   const logString = `Server received request from ${req.ip}:
```
Comment (joshxg) [inline src/server/middleware/middleware.js]: Maybe de structure these fields beforehand?

File: src/server/routes/masterlist.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+      res.json({ masterlists });
+    } catch (err) {
+      console.log(err);
+    }
+  } else {
+    res.statusCode = 401;
+    res.send(`Invalid JWT: ${token}.serialise()}`);
```
Comment (joshxg) [inline src/server/routes/masterlist.js]: I think method calls in here are a little ugly, personally. maybe const serializedToken = .. above would be nicer?

File: src/server/database/index.js (hunk 2 above)
Comment (wlthomson) [reply src/server/database/index.js]: I'll squash the newlines, but will leave the comment for now as it is related to #77.

File: src/server/routes/masterlist.js (hunk 4 above)
Comment (wlthomson) [reply src/server/routes/masterlist.js]: This is fixed in #139.

File: src/server/middleware/middleware.js (hunk 3 above)
Comment (wlthomson) [reply src/server/middleware/middleware.js]: Surprised that eslint didn't have a fit about that. On the one hand I like it explicit, as it makes the log string easy to read, but destructuring is tidier.

File: src/server/test/test.js [hunk 5]
```diff
@@ -64,3 +66,18 @@ export const getTestRecords = () => {
...
+
+// TODO: remove test records
+export const createTestRecords = async db => {
+  const testRecords = getTestRecords();
+
+  console.log(`Creating ${testRecords.length} test records...`, '\n');
+
+  try {
+    await db.insertRecords(testRecords);
+    // eslint-disable-next-line no-console
```
Comment (joshxg) [inline src/server/test/test.js]: Already have a file wide disable for this one

File: src/server/database/index.js (hunk 2 above)
Comment (wlthomson) [reply src/server/database/index.js]: Fixed.

File: src/server/middleware/middleware.js (hunk 3 above)
Comment (wlthomson) [reply src/server/middleware/middleware.js]: Fixed.

File: src/server/test/test.js (hunk 5 above)
Comment (wlthomson) [reply src/server/test/test.js]: Fixed.

**Comment by wlthomson** [comment]
All comments and conflicts resolved. @joshxg Ready to merge?

File: src/server/middleware/middleware.js [hunk 6]
```diff
@@ -4,22 +4,27 @@ import cors from 'cors';
 import JWT from '../../utilities/authorisation/JWT';
```
Comment (joshxg) [inline src/server/middleware/middleware.js]: Just wondering if you want to get rid of more of these new lines in this file or if you like them?

**Comment by wlthomson** [comment]
> @wlthomson Awesome. Looks heaps better! Just wanna say I prefer request, response and error over the shortened versions, but I think I might be alone in this one. Oh also just a new conflict :(

I agree, explicit is best. This is probably a repo-wide thing, so I'll make an issue for it.

File: src/server/middleware/middleware.js (hunk 6 above)
Comment (wlthomson) [reply src/server/middleware/middleware.js]: I think in this case the newlines are useful for separating the logic into "chunks". We can probably have a style discussion when we have time about consistent newlines as this is something that prettier and eslint cant do for us :(.

**Comment by wlthomson** [comment]
@joshxg All ready. Accidentally merged development before pulling it so... sorry about that :(.

**Pull request merged**